incubator/gocd/templates/gocd-agent-deployment.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "gocd.fullname" . }}-agent
+  labels:
+    app: {{ template "gocd.name" . }}-agent
+    chart:  "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+    component: {{ template "gocd.name" . }}-agent
```

Comment:
Don't include the name template in components. That's in the `app` label already.

`component: agent`

Please fix it throughout the chart.